lms/djangoapps/custom_search/search.py

```diff
@@ -0,0 +1,19 @@
+"""
+Filters and Result handlers for custom search rules
+"""
+
+from lms.lib.courseware_search.lms_filter_generator import LmsSearchFilterGenerator
+
+
+class CustomSearchFilterGenerator(LmsSearchFilterGenerator):
```

Comment:
Sorry I should've been clearer. Can you move the custom search class to the same directory the `LmsSearchFilterGenerator` is, `lms/lib/courseware_search/lms_filter_generator.py`? 

Also, would it be possible to rename this filter generator to a name that's more descriptive regarding what the filter generator does, such as `CatalogInviteOnlySearchFilterGenerator`? Please note that I don't have any better suggestions, so if you have any other suggestion, please go with it instead.

---

Just for clarification, initially, my suggestion to create a django app was in order to utilize this filter in case the change was not upstreamed. But since we're working on upstreaming it, there's no need for creating a new django app for it.

Sorry, I missed it in the initial internal pull request you created.

Reply:
Oh, got it 👍 

I'm not sure if it's better but just to put it out there, there's another approach we can take as well:

There's currently a `SEARCH_SKIP_ENROLLMENT_START_DATE_FILTERING` options in the settings. We can add similar settings:
`SEARCH_SHOW_INVITATION_ONLY_COURSES`
`SEARCH_SHOW_NON_CATALOG_COURSES`

Then instead of creating a new `SearchFilterGenerator` we can modify the `LmsSearchFilterGenerator` to respect these settings. This way we'd be able to configure these options separately if needed, e.g, disabling invitation-only while still allowing non-catalog courses.

@nizarmah  Any thoughts on this?